Source/WebCore/html/PermissionsPolicy.h

```diff
@@ -84,17 +85,16 @@ class PermissionsPolicy {
     private:
         std::variant<HashSet<SecurityOriginData>, AllowAllOriginsTag> m_origins;
     };
-    Ref<SecurityOrigin> declaredOrigin(const HTMLIFrameElement&) const;
-    Allowlist parseAllowlist(StringView, const SecurityOriginData& containerOrigin, const SecurityOriginData& targetOrigin, bool useStarAsDefaultAllowlistValue);
-
     // https://w3c.github.io/webappsec-permissions-policy/#policy-directives
     using PolicyDirective = HashMap<Feature, Allowlist, IntHash<Feature>, WTF::StrongEnumHashTraits<Feature>>;
-    PolicyDirective parsePolicyDirective(StringView, const SecurityOriginData& containerOrigin, const SecurityOriginData& targetOrigin, bool useStarAsDefaultAllowlistValue);
+    static PermissionsPolicy::PolicyDirective processPermissionsPolicyAttribute(const HTMLIFrameElement&);
 
-    PolicyDirective m_effectivePolicy;
-};
+private:
+    bool computeInheritedPolicyValueInContainer(Feature, const HTMLFrameOwnerElement*, const SecurityOriginData&) const;
+    bool getFeatureValueForOrigin(Feature, Document&, const SecurityOriginData&) const;
 
-enum class LogPermissionsPolicyFailure : bool { No, Yes };
-extern bool isPermissionsPolicyAllowedByDocumentAndAllOwners(PermissionsPolicy::Feature, const Document&, LogPermissionsPolicyFailure = LogPermissionsPolicyFailure::Yes);
+    using InheritedPolicy = HashMap<Feature, bool, IntHash<Feature>, WTF::StrongEnumHashTraits<Feature>>;
+    InheritedPolicy m_inheritedPolicy;
```

Comment:
Why using a HashMap if it is always filled with all keys?

Reply:
I was just following the spec: https://w3c.github.io/webappsec-permissions-policy/#permissions-policy-inherited-policy.
Do you think it's better to change it?

Reply:
Yes, a HashMap adds complexity without bringing benefit.